### ISSUE TITLE
add option to skip  document validation

### ISSUE THREAD

**Issue by n1ru4l** [dotansimha/graphql-code-generator#2565]
**Describe the bug**

Relay style arguments result in errors during type generations

I think this is related to the more strict validations done with graphql-toolkit.

```gql
fragment UserLogin on User
  @argumentDefinitions(
    avatarHeight: { type: "Int", defaultValue: 10 }
    avatarWidth: { type: "Int", defaultValue: 10 }
  ) {
  id
  login
  avatar(width: $avatarWidth, height: $avatarHeight) {
    id
    url
  }
}
```

```graphql
query user {
  users {
    ...UserLogin @arguments(avatarHeight: 30, avatarWidth: 30)
  }
}
```

```
GraphQLDocumentError: Variable "$avatarWidth" is not defined by operation "user".
GraphQLDocumentError: Variable "$avatarHeight" is not defined by operation "user".
```

This worked in version prior to v.1.7.0

**Comment by n1ru4l** [comment]
https://github.com/dotansimha/graphql-code-generator/blob/2a64793ccffa40f56de4cea6db5f946d580f1560/packages/graphql-codegen-core/src/execute-plugin.ts#L53

@dotansimha Can we add an option to disable this validation?

**Comment by dotansimha** [comment]
Fixed in 1.8.0 🚀
